Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -438,13 +477,14 @@ void TextureMapperLayer::collectDamageSelf(TextureMapperPaintOptions& options, D
     // Inferred damage always takes precedence over any other kind of damage as it stores
     // the damage that is a result of the layer-level operations.
     if (!m_inferredGlobalDamage.isEmpty() || !m_inferredLayerDamage.isEmpty()) {
+        auto clipBounds = options.textureMapper.clipBounds();
```

Comment:
I was checking how expensive it was to copy what `clipBounds()` returned (it's cheap, it's an `IntRect`, although you could also do `const auto& clipBounds` to bind to it). 

Then I thought `why is `clipBounds()` not `const`? I looked at the implementation, and it does

```
 clipStack().current().scissorBox;
```

neither `clipStack()` nor `current()` are `const`. It makes sense to have `non-const` options as often they will be used to modify things. But I wonder if we should have const overloads, too, as some of the usages are read-only.

But anyway. Beyond the `const auto&` suggestion, the rest is outside of the scope of this PR

Reply:
Noted down, thanks!